Automattic/wp-calypso
# Checkout: Move PaymentChatButton to new checkout

#### Changes proposed in this Pull Request

The `PaymentChatButton` component from old checkout is used by `WPCheckoutOrderSummary`. This PR moves it to new checkout since old checkout is deprecated and will soon be removed.

Fixes https://github.com/Automattic/wp-calypso/issues/47115

![paymentchatbutton](https://user-images.githubusercontent.com/2036909/99292922-e82fcf80-280f-11eb-8b00-6a2cb69d03d2.png)


#### Testing instructions

Testing this is difficult, because it requires a chat to be available. You can check that the chat loads correctly by applying the following patch on top of this PR and then loading checkout with a business plan in your cart. You should see the button in the sidebar (or in the header at small widths).

```diff
diff --git a/client/components/happychat/button.jsx b/client/components/happychat/button.jsx
index 3023283bec..98eb527d0d 100644
--- a/client/components/happychat/button.jsx
+++ b/client/components/happychat/button.jsx
@@ -107,7 +107,7 @@ export default connect(
        ( state ) => ( {
                hasUnread: hasUnreadMessages( state ),
                getAuth: getHappychatAuth( state ),
-               isChatAvailable: isHappychatAvailable( state ),
+               isChatAvailable: true,
                isChatActive: hasActiveHappychatSession( state ),
                isConnectionUninitialized: isHappychatConnectionUninitialized( state ),
        } ),
diff --git a/client/jetpack-connect/happychat-button.jsx b/client/jetpack-connect/happychat-button.jsx
index 1cbfd8bbb3..2e859b0dd1 100644
--- a/client/jetpack-connect/happychat-button.jsx
+++ b/client/jetpack-connect/happychat-button.jsx
@@ -61,7 +61,7 @@ JetpackConnectHappychatButton.propTypes = {
 };

 export default connect( ( state ) => ( {
-       isChatAvailable: isHappychatAvailable( state ),
+       isChatAvailable: true,
        isChatActive: hasActiveHappychatSession( state ),
        isLoggedIn: Boolean( getCurrentUserId( state ) ),
 } ) )( localize( JetpackConnectHappychatButton ) );
diff --git a/client/my-sites/checkout/composite-checkout/components/wp-checkout-order-summary.js b/client/my-sites/checkout/composite-checkout/components/wp-checkout-order-summary.js
index 6692c65a60..2c4f5cea6f 100644
--- a/client/my-sites/checkout/composite-checkout/components/wp-checkout-order-summary.js
+++ b/client/my-sites/checkout/composite-checkout/components/wp-checkout-order-summary.js
@@ -322,7 +322,7 @@ export function CheckoutSummaryHelp( { isMonthlyPricingTest = false } ) {
        const supportVariationDetermined = useSelector( isSupportVariationDetermined );
        const supportVariation = useSelector( getSupportVariation );

-       const happyChatAvailable = useSelector( isHappychatAvailable );
+       const happyChatAvailable = true;
        const presalesChatAvailable = useSelector( isPresalesChatAvailable );
        const presalesEligiblePlanLabel = getHighestWpComPlanLabel( plans );
        const isPresalesChatEligible = presalesChatAvailable && presalesEligiblePlanLabel;
```
